### PR TITLE
Improve formatting with tab characters

### LINKED ISSUE #251: tabs are not preserved in string literals
Tabs within strings are converted to two spaces. Similar code formatters like rubocop and Prettier preserve tabs within strings, making this behavior surprising.

I noticed this while running nixpkgs-fmt on a NixOS configuration file with rules for /etc/exports. The change doesn't affect the meaning of the file, but might for other file types (TSV?).

## Input
```nix
{
  simpleStringLiteral = "col	col";

  multilineStringLiteral = ''
    col	col	col
  '';
}
```

## Output
```nix
--- Input	
+++ Output	
@@ -1,7 +1,7 @@
 {
-  simpleStringLiteral = "col	col";
+  simpleStringLiteral = "col  col";
 
   multilineStringLiteral = ''
-    col	col	col
+    col  col  col
   '';
 }
```

## Desired output

```nix
{
  simpleStringLiteral = "col	col";

  multilineStringLiteral = ''
    col	col	col
  '';
}
```

### PR DESCRIPTION
**Pull request by mohe2015**
Closes #251

### CODE REVIEW

File: test_data/indent_tabs-3.good.nix [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# 	this	is cool
+{
+  # nix doesn't see tabs as indentation so don't replace them
+  test = ''
+    	test
+    	test
+  '';
```
Comment (mohe2015) [inline test_data/indent_tabs-3.good.nix:7]: The current parser thinks the tab character belongs to the indentation...

Comment (mohe2015) [reply test_data/indent_tabs-3.good.nix:7]: No, I'm just stupid

**Comment by zimbatm** [comment]
I agree with where this is going. This would be the ideal behaviour.

**Comment by mohe2015** [comment]
To be completely honest I don't think I will find the motivation soon to keep working on this. Anyone feel free to take over.

**Comment by mohe2015** [comment]
bad:
image
good:
image

bad:
image
good:
image

bad:
image
image

**Comment by mohe2015** [comment]
@zimbatm If you find the time to look at this again that would be great. I added some testcases that (see screenshots) show that most tabs are correctly removed. I'm currently not interested in fixing the remaining cases. The diff on nixpkgs in https://github.com/nix-community/nixpkgs-fmt/pull/270#issuecomment-974707853 should still be valid but I would be fine to verify this again before merging.

**Comment by zimbatm** [comment]
thanks!

**Pull request merged**